examples/render_grid/run.py

```diff
@@ -0,0 +1,334 @@
+            )
+        ]
+    )
+    spear.configure_system(config=config)
+    instance = spear.Instance(config=config)
+
+    # initialize actors and components
+    with instance.begin_frame():
+        # find functions
+        AActor = instance.unreal_service.get_static_class(class_name="AActor")
+        AActor_K2_SetActorLocation = instance.unreal_service.find_function_by_name(
+            uclass=AActor, function_name="K2_SetActorLocation"
+        )
+        AActor_K2_SetActorRotation = instance.unreal_service.find_function_by_name(
+            uclass=AActor, function_name="K2_SetActorRotation"
+        )
+
+        UGameplayStatics = instance.unreal_service.get_static_class(
+            class_name="UGameplayStatics"
+        )
+        UGameplayStatics_GetPlayerController = (
+            instance.unreal_service.find_function_by_name(
+                uclass=UGameplayStatics, function_name="GetPlayerController"
+            )
+        )
+
+        USpSceneCaptureComponent2D = instance.unreal_service.get_static_class(
+            class_name="USpSceneCaptureComponent2D"
+        )
+
+        USpSceneCaptureComponent2D_Initialize = (
+            instance.unreal_service.find_function_by_name(
+                uclass=USpSceneCaptureComponent2D,
+                function_name="Initialize",
+            )
+        )
+        USpSceneCaptureComponent2D_Finalize = (
+            instance.unreal_service.find_function_by_name(
+                uclass=USpSceneCaptureComponent2D,
+                function_name="Terminate",
+            )
+        )
+
+        UGameplayStatics_default_object = instance.unreal_service.get_default_object(
+            uclass=UGameplayStatics, create_if_needed=False
+        )
+
+        BP_Camera_Sensor = instance.unreal_service.load_object(
+            class_name="UClass",
+            outer=0,
+            name="/SpComponents/Blueprints/BP_Camera_Sensor.BP_Camera_Sensor_C",
+        )
+
+        # We will control the pose of the camera rig with a single parent actor
+        parent_actor = instance.unreal_service.spawn_actor_from_class(uclass=AActor)
+
+        USceneComponent = instance.unreal_service.get_static_class(
+            class_name="USceneComponent"
+        )
+
+        parent_root_component = (
+            instance.unreal_service.create_scene_component_by_class_from_actor(
+                scene_component_class=USceneComponent,
+                owner=parent_actor,
+                scene_component_name="DefaultSceneRoot",
+            )
+        )
+
+        bp_camera_sensor_actors = []
+        final_tone_curve_hdr_components = []
+        for i in range(rows):
+            for j in range(cols):
+                # These offsets define the spatial displacement of the cameras in the grid
+                # with respect to the parent actor's pose
+                offset = {"x": 0, "y": (j - cols / 2) * 20, "z": -(i - rows / 2) * 20}
```

Comment:
Would it be better in this example to define an array of offsets at the top of the file using a doubly nested loop, and from then on, just iterate over the array of offsets? That would confine the 2D grid assumption to a smaller part of the code.